portable/GCC/ARM_CR5/port.c

```diff
@@ -505,21 +595,23 @@ void FreeRTOS_Tick_Handler( void )
 }
 /*-----------------------------------------------------------*/
 
-#if( configUSE_TASK_FPU_SUPPORT != 2 )
+#if ( configUSE_TASK_FPU_SUPPORT == 1 )
 
     void vPortTaskUsesFPU( void )
     {
         uint32_t ulInitialFPSCR = 0;
 
-        /* A task is registering the fact that it needs an FPU context.  Set the
-         * FPU flag (which is saved as part of the task context). */
+        /*
+         * A task is registering the fact that it needs an FPU context. Set the
+         * FPU flag (which is saved as part of the task context).
+         */
         ulPortTaskHasFPUContext = pdTRUE;
 
         /* Initialise the floating point status register. */
         __asm volatile ( "FMXR  FPSCR, %0" ::"r" ( ulInitialFPSCR ) : "memory" );
```

Comment:
Does this initialization of the FPSCR need to happen somewhere else then if configUSE_TASK_FPU_SUPPORT is set to 2? I'm not sure if the register is set to zero by default and don't see it being done anywhere else?

Reply:
When `configUSE_TASK_FPU_SUPPORT`  is set to 2, initial context created in `pxPortInitialiseStack` sets every stored value of FP context (including FPSCR) to 0 which is then restored correctly in the context restore code.